File: src/lambda/python-with-layer/fizzbuzz.py
import json


def get_fizzbuzz(upper_limit):
    fizzbuzz = []

    for i in range(1, upper_limit + 1):
        if i % 15 == 0:
            fizzbuzz.append('fizzbuzz')
        elif i % 5 == 0:
            fizzbuzz.append('buzz')
        elif i % 3 == 0:
            fizzbuzz.append('fizz')
        else:
            fizzbuzz.append(str(i))

    return fizzbuzz


def handler(event):
    return json.dumps(get_fizzbuzz(event['upper_limit']))

File: src/lambda/node/fizzbuzz.ts
import { range } from '../../utils/range'
import { Callback, Context } from 'aws-lambda'

type FizzBuzzEvent = {
  upperLimit: number
}

const fizzBuzz = (upperLimit: number): string[] =>
  range(1, upperLimit).map((value) => {
    if (value % 15 === 0) {
      return 'fizzbuzz'
    } else if (value % 5 === 0) {
      return 'buzz'
    } else if (value % 3 === 0) {
      return 'fizz'
    } else {
      return `${value}`
    }
  })

export const handler = (
  event: FizzBuzzEvent,
  _context: Context,
  callback: Callback
): void => {
  const answer = fizzBuzz(event.upperLimit)
  callback(null, JSON.stringify(answer))
}
